File: app/src/main/java/com/ndmsystems/coala/resource_discovery/ResourceDiscoveryResult.java
package com.ndmsystems.coala.resource_discovery;

import java.net.InetSocketAddress;

/**
 * Created by bas on 19.09.16.
 */
public class ResourceDiscoveryResult {
    private final String resources;
    private final InetSocketAddress host;

    public ResourceDiscoveryResult(String resources, InetSocketAddress host) {
        this.resources = resources == null ? "" : resources;
        this.host = host;
    }

    public InetSocketAddress getHost() {
        return host;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ResourceDiscoveryResult that = (ResourceDiscoveryResult) o;

        if (!resources.equals(that.resources)) return false;
        return host.equals(that.host);

    }

    @Override
    public int hashCode() {
        int result = resources.hashCode();
        result = 31 * result + host.hashCode();
        return result;
    }
}

File: app/src/main/java/com/ndmsystems/coala/resource_discovery/ResourceDiscoveryResult.kt
package com.ndmsystems.coala.resource_discovery

import java.net.InetSocketAddress

/**
 * Created by bas on 19.09.16.
 */
class ResourceDiscoveryResult(payload: String, host: InetSocketAddress) {
    val payload: String
    val host: InetSocketAddress

    init {
        this.payload = payload
        this.host = host
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ResourceDiscoveryResult
        return if (payload != that.payload) false else host == that.host
    }

    override fun hashCode(): Int {
        var result = payload.hashCode()
        result = 31 * result + host.hashCode()
        return result
    }
}